### ISSUE TITLE
Mask mode totally busted right now

### ISSUE THREAD

**Issue by magnumripper** [magnumripper/JohnTheRipper#1041]
This must have been introduced by caeda0df
```
$ ../run/john -stdout -mask=?d
#
1p 0:00:00:00 100.00% (2015-02-02 20:38) 4.000p/s #

$ ../run/john -stdout -mask=?d?d
##
1p 0:00:00:00 100.00% (2015-02-02 20:38) 4.347p/s ##

$ ../run/john -stdout -mask=?d?d?d
##0
##1
##2
##3
##4
##5
##6
##7
##8
##9
10p 0:00:00:00 100.00% (2015-02-02 20:38) 43.47p/s ##9
```
This must be fixed immediately or we'll have to revert the GPU-mask merge. I thought that merge would not affect anything but raw-md4-opencl.

**Comment by kholia** [comment]
It would be awesome if we can detect such breakages early.

@jfoug is this something for the TS?

**Comment by magnumripper** [comment]
We have an issue for TS, but nothing in there yet.

**Comment by Sayantan2048** [comment]
:grimacing: I forgot to change some debug specific code. Sorry for the trouble. Fixed now in 6968fc3a86
